UnityProject/Assets/Scripts/ChemistryComponents/ReagentContainer.cs

```diff
@@ -199,7 +199,7 @@ private void OnImpact(UniversalObjectPhysics UOP, Vector2 Momentum)
 
 		private HashSet<Reaction> possibleReactions = new HashSet<Reaction>();
 		//Warning main thread only for now
-		public void ReagentsChanged()
+		public void ReagentsChanged(bool applyChange = true)
```

Comment:
is the idea for this new parameter to make some sort of dry run?

Reply:
Its basically so you can have two reactants mix without them reacting, and then manually make them react later. It's used by the blast yield detector to check wether a reaction it'll occur before it occurs as once it occurs you can no longer get any information about the reactants and quantities present, especially if it is like a smoke reaction and it empties the container.

Reply:
I imagine it might also be useful for other things down the line? but right now it is just for chemical grenades

Reply:
Okay